lib/system.nim

```diff
@@ -1200,6 +1200,14 @@ const
   hasSharedHeap = defined(boehmgc) or defined(gogc) # don't share heaps; every thread has its own
   taintMode = compileOption("taintmode")
 
+when defined(boehmgc):
```

Comment:
The problem is that `system/threads.nim` gets included before `system/mmdisp.nim`, and if I put it in `system/threads.nim`, it won't be available if `--threads:off` is selected. I also thought about putting it in a separate include file, but that would be wasteful. I'm open for an alternative suggestion; reordering the includes in `system.nim` seems to invariably break stuff.